yang/frr-zebra.yang

```diff
@@ -1992,12 +2052,67 @@ module frr-zebra {
         type uint32 {
           range "1..100000";
         }
+        units "megabits/sec";
         description
           "Link bandwidth informational parameter, in megabits.";
       }
       container link-params {
+        presence "Activates link parameters on this interface.";
         description
           "link-params for Traffic-Engineering (TE) use in IGP extensions.";
+        leaf metric {
+          type uint32;
+          description
+            "Link metric for MPLS-TE purpose.";
+        }
+        leaf max-bandwidth {
+          type rt-types:bandwidth-ieee-float32;
+          description
+            "Maximum bandwidth.";
+        }
+        leaf max-reservable-bandwidth {
+          type rt-types:bandwidth-ieee-float32;
+          description
+            "Maximum reservable bandwidth.";
+        }
+        container unreserved-bandwidths {
+          description
+            "All unreserved bandwidths.";
+          list unreserved-bandwidth {
+            key "priority";
+            leaf priority {
+              type uint8 {
+                range "0 .. 7";
+              }
+              description
+                "Priority from 0 to 7.";
+            }
+            leaf unreserved-bandwidth {
+              type rt-types:bandwidth-ieee-float32;
+              mandatory true;
```

Comment:
This is a TE attribute, but it's getting marked mandatory true for everything, this feels wrong. What if I'm not doing TE anything?

Are all these TE attributes `config true`?

Reply:
`unreserved-bandwidth` is not a presence container, it's a list. It can't be configured without the value. It is only mandatory when you create a list instance (`priority` is the key).